api/v1alpha3/conversion.go

```diff
@@ -113,20 +113,33 @@ func Convert_v1alpha3_OpenStackClusterSpec_To_v1alpha4_OpenStackClusterSpec(in *
 	return autoConvert_v1alpha3_OpenStackClusterSpec_To_v1alpha4_OpenStackClusterSpec(in, out, s)
 }
 
+// Convert_v1alpha4_OpenStackClusterSpec_To_v1alpha3_OpenStackClusterSpec has to be added by us because we have to
+// convert the Type of CloudsSecret from SecretReference to string.
+func Convert_v1alpha4_OpenStackClusterSpec_To_v1alpha3_OpenStackClusterSpec(in *v1alpha4.OpenStackClusterSpec, out *OpenStackClusterSpec, s conversion.Scope) error {
+	if in.IdentityRef != nil {
+		out.CloudsSecret.Name = in.IdentityRef.Name
```

Comment:
Couldn't the CloudSecret here be nil?
```suggestion
		out.CloudsSecret.Name = &corev1.SecretReference{Name: in.IdentityRef.Name}
```

Reply:
Sure, thanks